### PR TITLE
Feat auth server

### PR DESCRIPTION
**Pull request by balmas**
This PR requires the changes in alpheios-project/components#368 and alpheios-project/wordlist#15

It adds session and getEndpoints methods to the Authenticator interface, to support server-side auth and deployment-environment config of user data endpoints.

It also implements a SessionAuthenticator which can be used in place of the AppAuthenticator for traditional server-side web applications

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,13 +29,13 @@
   "homepage": "https://github.com/alpheios-project/wordsvc#readme",
   "devDependencies": {
     "alpheios-client-adapters": "github:alpheios-project/client-adapters",
-    "alpheios-components": "github:alpheios-project/components",
+    "alpheios-components": "file:../components",
```
Comment (irina060981) [inline package.json]: should be changed to github

File: package.json [hunk 2]
```diff
@@ -29,13 +29,13 @@
...
     "alpheios-data-models": "github:alpheios-project/data-models",
     "alpheios-experience": "github:alpheios-project/experience",
     "alpheios-inflection-tables": "git://github.com/alpheios-project/inflection-tables.git",
     "alpheios-node-build": "github:alpheios-project/node-build",
     "alpheios-res-client": "github:alpheios-project/res-client",
-    "alpheios-wordlist": "github:alpheios-project/wordlist",
+    "alpheios-wordlist": "github:alpheios-project/wordlist#issue-14",
```
Comment (irina060981) [inline package.json]: the branch reference should be removed

File: src/lib/session-authenticator.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  session () {
+    return new Promise((resolve,reject) => {
+      window.fetch(this.sessionUrl).then((resp) => {
+        if (! resp.ok) {
+          reject(resp.code)
+        } else {
+          resolve(resp.json())
+        }
+      })
+    })
```
Comment (kirlat) [inline src/lib/session-authenticator.js]: I'm wondering if and how the `session` promise gets resolved if `fetch` itself is rejected. Do we need to have a `catch` block for it?

File: src/lib/session-authenticator.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  getUserData () {
+    return new Promise((resolve,reject) => {
+      window.fetch(this.tokenUrl).then((resp) => {
+        if (! resp.ok) {
+          reject(resp.code)
+        } else {
+          resolve(resp.json())
+        }
+      })
+    })
```
Comment (kirlat) [inline src/lib/session-authenticator.js]: Same here

File: package.json (hunk 1 above)
Comment (balmas) [reply package.json]: oops, thanks will fix

File: package.json (hunk 2 above)
Comment (balmas) [reply package.json]: will fix thanks!

File: src/lib/session-authenticator.js (hunk 3 above)
Comment (balmas) [reply src/lib/session-authenticator.js]: good point, will fix

File: src/lib/session-authenticator.js (hunk 4 above)
Comment (balmas) [reply src/lib/session-authenticator.js]: will fix

**Comment by balmas** [comment]
> @balmas , I have uploaded this version and branches with
> [alpheios-project/wordlist#15](https://github.com/alpheios-project/wordlist/pull/15)
> [alpheios-project/components#368](https://github.com/alpheios-project/components/pull/368)
> 
> I used this link http://localhost:8888/demo/auth.html
> I have got the following errors on page start up (before login)
> 
> image

ah, I missed adding the (unimplemented) session method to the AppAuthenticator. will fix.

**Pull request merged**